pkg/signals/signals.go

```diff
@@ -0,0 +1,34 @@
+package signals
+
+import (
+	"os"
+	"os/signal"
+	"syscall"
+)
+
+// onlyOneSignalHandler is a channel that is closed the first time we invoke
+// trapSignals(). If trapSignals() is ever invoked again the application will
+// panic trying to close a channel that is already closed. This is to ensure
+// we're only trapping the singals in one place.
+var onlyOneSignalHandler = make(chan struct{})
+
+// Setup registers for SIGTERM and SIGINT. A channel is returned
+// which is closed on one of these signals. If a second signal is caught,
+// the program is terminated with exit code 1.
+func Setup() <-chan struct{} {
+	close(onlyOneSignalHandler) //panics when called twice
+
+	stop := make(chan struct{}) //create the channel we will return to the callee
+
+	c := make(chan os.Signal, 2)											// create channel c that will receive the os.Signals
```

Comment:
Personally I don't like inline comments at all, but if you want to keep these, I'd say only have one space between your code and the start of the comment, e.g.

```golang
c := make(chan os.Signal, 2) // create channel c that will receive the os.Signals
```